paywall/src/unlock.js/PostMessageEmitters/DataIframeMessageEmitter.ts

```diff
@@ -53,15 +42,33 @@ export default class DataIframeMessageEmitter extends FancyEmitter {
     const url = new URL(this.iframe.src)
     addIframeToDocument(window, this.iframe)
 
-    const { postMessage, addHandler } = mainWindowPostOffice(
-      window,
-      this.iframe,
+    const { postMessage } = emitPostMessagesFrom(
+      this.iframe.contentWindow,
       url.origin,
-      'main window',
-      'Data iframe'
+      window,
+      this.emitWrapper
     )
     this.postMessage = postMessage
-    this.addHandler = addHandler
+  }
+
+  emitWrapper = (type: PostMessages, payload?: any) => {
```

Comment:
This is a bit of a hack to match the previous behavior of the registered listeners -- in the future I think we should do as little manipulation of messages as possible. If we don't think it's useful to receive an update with an empty object of locks, the data iframe should never emit one. Likewise for web3 calls.

See: [the robustness principle](https://en.wikipedia.org/wiki/Robustness_principle)